src/gossip/service.zig

```diff
@@ -258,12 +260,40 @@ pub const GossipService = struct {
         deinitMux(&self.failed_pull_hashes_mux);
     }
 
-    /// these threads should run forever - so if they join - somethings wrong
-    /// and we should shutdown
-    fn joinAndExit(self: *Self, handle: *std.Thread) void {
-        handle.join();
-        self.exit.store(true, .unordered);
-    }
+    pub const RunHandles = struct {
+        exit: *AtomicBool,
+        receiver_thread: std.Thread,
+        packet_verifier_thread: std.Thread,
+        message_processor_thread: std.Thread,
+        message_builder_thread: ?std.Thread,
+        responder_thread: std.Thread,
+        dumper_thread: ?std.Thread,
+
+        /// If any of the threads join, all other threads will be signalled to join.
+        pub fn joinAndExit(handles: RunHandles) void {
```

Comment:
can you change this to follow the `Self` convention

Reply:
Small problem: using `const Self = @This()` both here and in the enclosing `GossipService` struct namespace means using the self convention here is an ambiguous reference. This is one of the big disadvantages of strictly following the `Self` convention, instead of directly using the type name. Either we make an exception here, or I move the `RunHandles` struct outside of `GossipService`. Which do you prefer?

Also note: this same issue exists for `VerifyMessageEntry`, which was also breaking this rule before the PR.

Reply:
ah i see - ok keeping `RunHandles`/`VerifyMessageEntry` as-is is ok then

Reply:
Understood, sending changes in a minute.